joblib/test/test_parallel.py

```diff
@@ -206,19 +202,17 @@ def test_parallel_pickling():
     """ Check that pmap captures the errors when it is passed an object
         that cannot be pickled.
     """
-    def g(x):
-        return x ** 2
-
     try:
-        # pickling a local function always fail but the exception
-        # raised is a PickleError for python <= 3.4 and AttributeError
-        # for python >= 3.5
-        pickle.dumps(g)
-    except Exception as exc:
-        exception_class = exc.__class__
-
-    assert_raises(exception_class, Parallel(),
-                  (delayed(g)(x) for x in range(10)))
+        raise Exception
+    except Exception:
+        ex_type, ex, non_picklable = sys.exc_info()
+
+    assert_raises((PicklingError, TypeError), delayed, non_picklable)
```

Comment:
I think this is better to keep the previous solution. Basically we want to make sure that the exception is the same whether it happens in the main process or in a worker process inside a Parallel call.